fuel-vm/src/tests/blockchain.rs

```diff
@@ -2261,3 +2261,444 @@ fn various_ldc_issues_poc() {
+    );
+
+    ldcv2_reason_helper(load_blob, MemoryOverflow);
+}
+
+#[test]
+fn ldcv2__fails_when_blob_size_overflows() {
+    ldcv2_reason_helper(
+        vec![
+            op::not(0x20, RegId::ZERO),
+            op::ldc(RegId::HP, RegId::ZERO, 0x20, 1),
+        ],
+        MemoryOverflow,
+    );
+}
+
+#[test]
+fn ldcv2__fails_wehn_blob_id_ends_beyond_max_ram() {
+    // Then deploy another contract that attempts to read the first one
+    let reg_a = 0x20;
+    let reg_b = 0x21;
+
+    // cover blob_id_end beyond max ram
+    let load_blob = vec![
+        op::move_(reg_a, RegId::HP),           // r[a] := $hp
+        op::xor(reg_b, reg_b, reg_b),          // r[b] := 0
+        op::ori(reg_b, reg_b, 12),             // r[b] += 12 (will be padded to 16)
+        op::ldc(reg_a, RegId::ZERO, reg_b, 1), // Load first two words from the contract
+    ];
+
+    ldcv2_reason_helper(load_blob, MemoryOverflow);
+}
+
+#[test]
+fn ldcv2_fails_when_blob_not_in_inputs() {
+    // Then deploy another contract that attempts to read the first one
+    let reg_a = 0x20;
+    let reg_b = 0x21;
+
+    // contract not in inputs
+    let load_blob = vec![
+        op::ldc(reg_a, RegId::ZERO, reg_b, 1), // Load first two words from the contract
+    ];
+
+    ldcv2_reason_helper(load_blob, PanicReason::BlobNotFound);
+}
+
+#[test]
+fn load_blob_code__copies_expected_bytes() {
+    let mut test_context = TestBuilder::new(2322u64);
+    let gas_limit = 1_000_000;
+
+    let program_ops = vec![
+        op::movi(0x10, 0x11),
+        op::movi(0x11, 0x2a),
+        op::add(0x12, 0x10, 0x11),
+        op::log(0x10, 0x11, 0x12, 0x00),
+        op::ret(0x20),
+    ];
+
+    let program = program_ops.clone().into_iter().collect::<Vec<u8>>();
+    let blob_size = program.len();
+    let blob_id = BlobId::compute(&program);
+    test_context.setup_blob(program.clone());
+
+    let (script, _) = script_with_data_offset!(
+        data_offset,
+        vec![
+            op::movi(0x20, data_offset as Immediate18),
+            op::add(0x11, RegId::ZERO, 0x20),
+            op::movi(0x12, 0 as Immediate18),
+            op::movi(0x13, blob_size as Immediate18),
+            op::move_(0x22, RegId::SSP),
+            op::ldc(0x11, 0x12, 0x13, 1),
+            op::addi(0x21, 0x20, ContractId::LEN as Immediate12),
+            op::meq(0x30, 0x21, 0x22, 0x13),
+            op::ret(0x30),
+        ],
+        TxParameters::DEFAULT.tx_offset()
+    );
+
+    let mut script_data = blob_id.to_vec();
+    script_data.extend(program.as_slice());
+
+    let result = test_context
+        .start_script(script, script_data)
+        .script_gas_limit(gas_limit)
+        .fee_input()
+        .execute();
+
+    let receipts = result.receipts();
+    let ret = receipts
+        .first()
+        .expect("A `RET` opcode was part of the program.");
+
+    assert_eq!(1, ret.val().expect("Return value"));
+}
+
+#[test]
+fn load_blob_code__fails_when_blob_offset_is_over_length() {
```

Comment:
Hmm, this test doesn't fail=) Maybe we need to change the name

Reply:
Renamed in 7ebac9e5. I'd like to have a bit more exact test for this. This should be improved elsewhere as well, I'll do it as part of https://github.com/FuelLabs/fuel-vm/issues/746